Framework/WorkflowAlgorithms/src/QENSFitSequential.cpp

```diff
@@ -0,0 +1,694 @@
+#include "MantidWorkflowAlgorithms/QENSFitSequential.h"
+
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/Axis.h"
+#include "MantidAPI/CompositeFunction.h"
+#include "MantidAPI/CostFunctionFactory.h"
+#include "MantidAPI/FunctionProperty.h"
+#include "MantidAPI/IFunction.h"
+#include "MantidKernel/BoundedValidator.h"
+#include "MantidKernel/ListValidator.h"
+#include "MantidKernel/MandatoryValidator.h"
+
+#include <boost/cast.hpp>
+#include <boost/regex.hpp>
+
+#include <sstream>
+#include <stdexcept>
+#include <unordered_map>
+
+namespace {
+Mantid::Kernel::Logger g_log("QENSFitSequential");
+
+using namespace Mantid::API;
+using namespace Mantid::Kernel;
+
+MatrixWorkspace_sptr convertSpectrumAxis(MatrixWorkspace_sptr inputWorkspace,
+                                         const std::string &outputName) {
+  auto convSpec = AlgorithmManager::Instance().create("ConvertSpectrumAxis");
+  convSpec->setLogging(false);
+  convSpec->setProperty("InputWorkspace", inputWorkspace);
+  convSpec->setProperty("OutputWorkspace", outputName);
+  convSpec->setProperty("Target", "ElasticQ");
+  convSpec->setProperty("EMode", "Indirect");
+  convSpec->execute();
+  // Attempting to use getProperty("OutputWorkspace") on algorithm results in a
+  // nullptr being returned
```

Comment:
This appears to be rather strange behaviour, why would the OutputWorkspace property be set to a nullptr, since the Workspace still exists somewhere and it seems reasonable to extract it from the algorithm and not worry about the ADS at all.